### ISSUE TITLE
Libreoffice windows stop working

### ISSUE THREAD

**Issue by scossu** [swaywm/sway#5262]
After a first launch and little interaction, LibreOffice is unable to open windows and leaves open processes. 

I have replicated this with both native Wayland and XWayland. Examples below are for native Wayland.

Sway version: 1.4

LibreOffice version: 6.3.5-4

Sway run script (from tty):

```

#!/bin/sh
export _JAVA_AWT_WM_NONREPARENTING=1

export MOZ_ENABLE_WAYLAND=1

export QT_QPA_PLATFORM=wayland-egl
export QT_QPA_PLATFORMTHEME=qt5ct
export QT_WAYLAND_FORCE_DPI=96
export QT_WAYLAND_DISABLE_WINDOWDECORATION=1

export SDL_VIDEODRIVER=wayland

SAL_USE_VCLPLUGIN=gtk3

export TERMINAL=kitty
export XKB_DEFAULT_LAYOUT=us

exec sway -d 2>~/.sway.log
```

Actions: 

- Start Sway
- Remove `~/.config/libreoffice/4`
- Launch `soffice` from terminal
- App starts, but outputs `Failed to open display` on terminal
- Ctrl-O
- Open file dialog appears. App outputs `(soffice:30657): Gtk-WARNING **: 21:01:44.063: Failed to measure available space: No route to host` on console
- Select a CSV file. LO opens another dialog / preview window. 
- Press "OK" button
- Application outputs `(soffice:30657): Gtk-WARNING **: 21:16:45.765: gtk_widget_size_allocate(): attempt to allocate widget with width -3 and height -3` on terminal
- Spreadsheet appears next to treminal window. 
- Ctrl-O again
- Open another CSV file
- Another preview window appears
- Click OK
- No spreadsheet opens
- Close application
- Restart again from terminal with `soffice` 
- Splash appears and hangs
- App outputs the "attempt to allocate widget..." message on terminal several times

The only way to recover the application is to delete `~/.config/libreoffice/4` every time.

**Comment by apiraino** [comment]
Hi @scossu I'm on debian bullseye and can't start libreoffice 7rc on Sway 1.5 with `xwayland disabled` in my config (actually I don't have X11 nor gnome installed at all). I have set your env vars, too.

Do you still experience this issue? In your issue you wrote "native Wayland" - are you 100% you were not under xwayland? I'm asking because with my 100% wayland+sway I only get an error that soffice.bin can't open the X11 display (pasting here for search engines)
```
$ soffice --version
LibreOffice 7.0.4.2 00(Build:2)

$ soffice 
/usr/lib/libreoffice/program/soffice.bin X11 error: Can't open display: 
   Set DISPLAY environment variable, use -display option
   or check permissions of your X-Server
   (See "man X" resp. "man xhost" for details)
```

I'm inverstigating, likely missing some other _undocumented_ env var or trick ...

 thanks for any info you may have

**Comment by apiraino** [comment]
confirmed that without `xwayland` libreoffice doesn't work on Sway

**Comment by ngortheone** [comment]
I am on gentoo, and libreoffice is compiled with Qt and without gtk support. LibreOffice indeed does not start without xwayland.
Going to rebuild it with gtk support and see if changes anything...

**Comment by apiraino** [comment]
> LibreOffice 7.0.4.2 with GTK3 seems to be working fine, out of the box.

yes, my understanding is that GTK3 fixed native Wayland compatibility (as in: "xwayland should not be needed anymore") I've tested this with Gimp 2.99.

FWIW I've installed the vanilla Libreoffice (not sure the issue reporter was using) and they work fine in my installation :man_shrugging:

**Comment by savchenko** [comment]
> FWIW I've installed the vanilla Libreoffice (not sure what the issue reporter was using) and it works fine on my installation man_shrugging

By "vanilla", do you mean one from https://www.libreoffice.org/download/download/ ? If yes, have you reported the issue to Debian package tracker?

**Comment by savchenko** [comment]
Debian Bullseye, running fine:
```
Version: 7.0.4.2
Build ID: 00(Build:2)
CPU threads: 12; OS: Linux 5.10; UI render: default; VCL: gtk3
Locale: en-AU (en_AU.UTF-8); UI: en-US
Debian package version: 1:7.0.4-3
Calc: threaded
```
Recipe:
1. `apt install libreoffice libreoffice-gtk3`
2. Add `xwayland disable` to Sway config
3. Cherry-pick relevant environment variables from https://github.com/savchenko/debian/blob/bullseye/roles/dotfiles/files/profile

**Comment by savchenko** [comment]
@emersion, shall this be closed?..
